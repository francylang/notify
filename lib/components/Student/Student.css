.values-classroom {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  margin: 10px 50px 47px 50px; }

.return-to-students {
  text-decoration: none; }

.return-to {
  margin-left: 5px; }

.return-to:hover {
  border-bottom: 2px solid #E65C4F; }

.student-link-students {
  display: flex;
  text-decoration: none;
  color: #888888;
  margin: 0 40px 0 45px; }

.core-values-ctnr {
  display: flex;
  flex-wrap: wrap;
  margin: 36px 12px 0 103px; }

.icon-left-arrow {
  margin-right: 10px; }

.student-name-header {
  color: #888888;
  font-size: 50px;
  font-weight: lighter;
  margin: 243px 0px 0 122px; }

.student-name-header:focus {
  outline: #d35656; }
