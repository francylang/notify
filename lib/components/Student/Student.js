import React, {Component} from 'react';
import PropTypes from 'prop-types';
import CoreValue from '../CoreValue/CoreValue.js';
import { Link } from 'react-router-dom';
import textFetch from '../../utils/textFetch.js';

class Student extends Component {
  constructor() {
    super();
    this.colors = ['red', 'green', 'blue', 'purple', 'yellow'];
  }

  async updateCoreValueAndTextParent(student, coreValue, studentId) {
    console.log('update and text:', student);

    this.props.plusOneCoreValue(coreValue, studentId);
    await textFetch(student, coreValue);
  }

  renderCoreValue() {

    const studentId = parseInt(this.props.match.params.id);
    const coreValues = this.props.coreValues[studentId];
    const student = this.props.students[studentId].name;
    console.log('render:', student);
    console.log('this:', this.props.students);



    return Object.keys(coreValues).map((coreValue, index) => {
      return (
        <CoreValue
          key={ coreValue }
          coreValue={ coreValue }
          plusOneCoreValue={ () =>
            this.updateCoreValueAndTextParent(student, coreValue, studentId)
          }
          studentId={ studentId }
          color={this.colors[index]}
          count={ coreValues[coreValue] }
          />
        );
    });
  }

  render() {
    const studentId = parseInt(this.props.match.params.id);
    return (
      <div>
        <h2 className='student-name-header'>{this.props.students[studentId].name}</h2>
          <div
            className='core-values-ctnr'>
            { this.renderCoreValue() }
          </div>
          <div className='return-to-students'>
            <div className='values-classroom'>
              <Link
                className='student-link-students'
                to={'/'}>
                <img
                  src='../lib/assets/leftArrow.svg'
                  className='icon-left-arrow'
                  width='25px'
                  height='25px' />
                <h3 className='return-to'>Return to Students</h3>
              </Link>
            </div>
          </div>
      </div>
    );
  }
}

Student.propTypes = {
  match: PropTypes.object,
  plusOneCoreValue: PropTypes.func,
  coreValues: PropTypes.object,
  students: PropTypes.array
};

export default Student;
