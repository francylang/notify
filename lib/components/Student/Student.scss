.values-classroom {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  margin: 10px 50px 47px 50px;
}

.return-to-students {
  text-decoration: none;
}

.return-to {
  margin-left: 5px;
}

.return-to:hover {
  border-bottom: 2px solid #49A6C3;
}

.student-link-students {
  display: flex;
  text-decoration: none;
  color: gray;
  margin: 0 40px 0 4px;
}

.core-values-ctnr {
  display: flex;
  flex-wrap: wrap;
  margin: 36px 12px 0 80px;
}

.icon-left-arrow {
  margin-right: 10px;
}

.student-name-header {
  color: gray;
  font-size: 50px;
  font-weight: lighter;
  margin: 242px 0px 0 90px;
}
