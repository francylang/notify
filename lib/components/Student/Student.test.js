import React from 'react';
import Student from './Student.js';
import { shallow } from 'enzyme';

describe('<Student />', () => {

  let wrapper;
  let mockFn = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Student studentId={5} />);
  });
  it('should exist', () => {

    expect(wrapper).toBeDefined();
  });

  it('should match the Student snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });
});
