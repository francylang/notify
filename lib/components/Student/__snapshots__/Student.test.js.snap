// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Student /> should match the Student snapshot 1`] = `undefined`;
